pkg/karmadactl/deinit.go

```diff
@@ -0,0 +1,345 @@
+	KarmadaDataPath string
+	Namespace       string
+	KubeConfig      string
+	KubeClientSet   *kubernetes.Clientset
+}
+
+// NewCmdDeInit remove karmada on kubernetes
+func NewCmdDeInit(cmdOut io.Writer, parentCommand string) *cobra.Command {
+	opts := CommandDeInitOption{}
+	cmd := &cobra.Command{
+		Use:          "deinit",
+		Short:        "remove karmada on kubernetes",
+		Long:         "remove karmada on kubernetes",
+		Example:      deInitExample(parentCommand),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.RunDeInit(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	flags := cmd.PersistentFlags()
+	flags.StringVarP(&opts.Namespace, "namespace", "n", "karmada-system", "Kubernetes namespace")
+	flags.StringVar(&opts.KubeConfig, "kubeconfig", filepath.Join(homeDir(), ".kube", "config"), "absolute path to the kubeconfig file")
+	flags.StringVarP(&opts.KarmadaDataPath, "karmada-data", "d", "/etc/karmada", "karmada data path. kubeconfig cert and crds files")
+	flags.BoolVarP(&opts.Force, "force", "f", false, "skip delete confirmation")
+	flags.BoolVarP(&opts.Recursive, "recursive", "r", false, "delete recursively, delete all resources in the namespace, including persistent storage.")
+	return cmd
+}
+func deInitExample(parentCommand string) string {
+	return fmt.Sprintf(`
+# Remove Karmada in Kubernetes cluster.
+%s deinit
+
+# Skip the prompt and just delete Karmada.
+%s deinit --force
+
+# Recursively delete, delete the namespace and all resources under the namespace.
+%s deinit --recursive
+`, parentCommand, parentCommand, parentCommand)
+}
+
+// Complete the conditions required to be able to run deinit.
+func (d *CommandDeInitOption) Complete() error {
+	restConfig, err := utils.RestConfig(d.KubeConfig)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("kubeconfig file: %s, kubernetes: %s\n", d.KubeConfig, restConfig.Host)
+	clientSet, err := utils.NewClientSet(restConfig)
+	if err != nil {
+		return err
+	}
+	d.KubeClientSet = clientSet
+	if _, err := d.KubeClientSet.CoreV1().Namespaces().Get(context.TODO(), d.Namespace, metav1.GetOptions{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate Check that there are enough flags to run the command.
+func (d *CommandDeInitOption) Validate() error {
+	if !isExist(d.KubeConfig) {
+		return fmt.Errorf("%q file does not exist.absolute path to the kubeconfig file", d.KubeConfig)
+	}
+
+	return nil
+}
+
+func isExist(fileName string) bool {
+	_, err := os.Stat(fileName)
+	if err != nil && os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+// DeleteDeployments Delete deployment by label LabelSelector
+func (d *CommandDeInitOption) DeleteDeployments() error {
```

Comment:
I guess the `DeleteDeployments` is beyond the scope of `CommandDeInitOption` whose responsibility is holding the options(not deleting resources).

I mean we can add a separated function to do the deleting things.